src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs

```diff
@@ -16,7 +17,9 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
         {
             if (bindingContext.ModelType == typeof(CancellationToken))
             {
-                var model = bindingContext.OperationBindingContext.HttpContext.RequestAborted;
+                // Force boxing so we can maintain a reference to this token in validation state.
```

Comment:
I agree you're fixing things here. I'm asking for a bit more in the comment because it didn't help me understand the issue i.e. what shouldn't be reverted later.